orchestrator/test_runner/src/happy_path.rs

```diff
@@ -109,18 +109,25 @@ pub async fn happy_path_test(
 }
 
 pub async fn wait_for_nonzero_valset(web30: &Web3, gravity_address: EthAddress) {
-    let start = Instant::now();
-    let mut current_eth_valset_nonce = get_valset_nonce(gravity_address, *MINER_ADDRESS, &web30)
-        .await
-        .expect("Failed to get current eth valset");
-
-    while 0 == current_eth_valset_nonce {
-        info!("Validator set is not yet updated to >0, waiting");
-        current_eth_valset_nonce = get_valset_nonce(gravity_address, *MINER_ADDRESS, &web30)
-            .await
-            .expect("Failed to get current eth valset");
-        delay_for(Duration::from_secs(4)).await;
-        if Instant::now() - start > TOTAL_TIMEOUT {
+    match tokio::time::timeout(TOTAL_TIMEOUT, async {
+        let mut current_eth_valset_nonce =
+            get_valset_nonce(gravity_address, *MINER_ADDRESS, &web30)
+                .await
+                .expect("Failed to get current eth valset");
+        while 0 == current_eth_valset_nonce {
+            info!("Validator set is not yet updated to >0, waiting");
+            current_eth_valset_nonce = get_valset_nonce(gravity_address, *MINER_ADDRESS, &web30)
+                .await
+                .expect("Failed to get current eth valset");
+            tokio::time::sleep(std::time::Duration::from_secs(4)).await;
```

Comment:
Same `use` statement stuff around `delay_for` and `Duration` in `happy_path` and `happy_path_v2` as in the earlier files in the changeset, just noting it here once rather than highlighting each instance.

Reply:
This commit: https://github.com/PeggyJV/gravity-bridge/pull/274/commits/fc227ac6a93fd02bdf80f92a525e1f38b163b906 fixes this, thank you.

Reply:
Ah, what I meant was rather that we can include `use std::time::Duration` and then when we call sleep, just refer to it like `tokio::time::sleep(Duration::from_secs(...))`, since there is no ambiguity to require fully qualifying it inline.